packages/lib-classifier/src/plugins/drawingTools/experimental/components/LineControls/LineControls.js

```diff
@@ -33,12 +32,20 @@ const LineControls = forwardRef(function LineControls({
   const OUTER_RADIUS = 40 / scale
   const INNER_RADIUS = 20 / scale
 
-  const { viewBox, rotate, width, height } = useContext(SVGContext)
-  const [x, y, w, h] = viewBox.split(' ').map(n => parseInt(n, 10))
-  const p = (activePosition == 'tr')
-    ? { x: x + (50 / scale), y: y + (50 / scale) }
-    : { x: x + w - (50 / scale), y: y + (50 / scale) }
+  const { canvas, rotate } = useContext(SVGContext)
+
+  // Get the transformation that's been applied to a parent element
+  const { matrix } = canvas.parentElement.parentElement.transform.baseVal.consolidate()
+  const { a: hScale, d: vScale, e: hTranslate, f: vTranslate } = matrix
+  const LINE_CONTROL_INSET = 70;
+
+  const { width, height } = canvas.getBBox()
+  const x = (LINE_CONTROL_INSET - hTranslate) / hScale
+  const y = (LINE_CONTROL_INSET - vTranslate) / vScale
 
+  const p = (activePosition === 'tl')
+    ? { x, y }
+    : { x: x + ((width - (LINE_CONTROL_INSET * 2)) / hScale), y }
```

Comment:
Maybe add a comment here to explain why you're subtracting the inset twice: once to get back to the right-hand edge of the image, then again to inset inside the image itself.